vdirsyncer/storage/dav.py

```diff
@@ -317,6 +349,39 @@ def request(self, method, path, **kwargs):
             url = utils.compat.urlparse.urljoin(self.url, path)
         if self._session is None:
             self._session = requests.session()
+            if self._use_oauth2_google:
+                self._session = OAuth2Session(
+                    client_id=oauth2_client_id,
+                    token=self._token,
+                    redirect_uri='urn:ietf:wg:oauth:2.0:oob',
+                    scope=['https://www.googleapis.com/auth/calendar',
+                           'https://www.googleapis.com/auth/carddav'],
+                    auto_refresh_url=OAUTH2_GOOGLE_REFRESH_URL,
+                    auto_refresh_kwargs={
+                        'client_id': oauth2_client_id,
+                        'client_secret': oauth2_client_secret,
+                    },
+                    token_updater=self._save_oauth2_token
+                )
+                if not self._token:
+                    authorization_url, state = self._session.authorization_url(
+                        OAUTH2_GOOGLE_TOKEN_URL,
+                        # access_type and approval_prompt are Google specific
+                        # extra parameters.
+                        access_type="offline", approval_prompt="force")
+                    print("Please go to %s" % authorization_url)
```

Comment:
Please use `click.echo` here, `print` will behave funky in multithread environments.

See https://github.com/pimutils/vdirsyncer/blob/master/vdirsyncer/storage/remotestorage.py#L96 for similar code.